Ark.Tools.Sql/AbstractSqlContextAsync.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Data;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NodaTime;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Ark.Tools.Sql
+{
+    public abstract class AbstractSqlContextAsync<Tag> : ISqlContextAsync
+    {
+        private DbConnection? _connection;
+        private DbTransaction? _transaction;
+        private IsolationLevel _isolationLevel;
+        private IDbConnectionManager? _connectionManager;
+
+        public IDbConnectionManager? ConnectionManager { get { return _connectionManager; } set { _connectionManager = value; } }
```

Comment:
connection is managed by the Factory, why is still here? Also is not initialized in the ctor.